source/mod_units.f90

```diff
@@ -122,37 +122,40 @@ subroutine units_openUnit(unit,fileName,formatted,mode,err,status,recl)
   units_uList(units_nList)%recl      = fRecl
   units_uList(units_nList)%open      = .true.
 
+  err = .false.
   if(formatted) then
     if(fRecl > 0) then
       if(fFio) then
         open(unit,file=fFileName,form="formatted",status=fStatus,iostat=ioStat,&
-          action=fAction,position=fPosition,round="nearest",recl=fRecl)
+          action=fAction,position=fPosition,round="nearest",recl=fRecl,err=10)
       else
         open(unit,file=fFileName,form="formatted",status=fStatus,iostat=ioStat,&
-          action=fAction,position=fPosition,recl=fRecl)
+          action=fAction,position=fPosition,recl=fRecl,err=10)
       end if
     else
       if(fFio) then
         open(unit,file=fFileName,form="formatted",status=fStatus,iostat=ioStat,&
-          action=fAction,position=fPosition,round="nearest")
+          action=fAction,position=fPosition,round="nearest",err=10)
       else
         open(unit,file=fFileName,form="formatted",status=fStatus,iostat=ioStat,&
-          action=fAction,position=fPosition)
+          action=fAction,position=fPosition,err=10)
       end if
     end if
   else
     open(unit,file=fFileName,form="unformatted",status=fStatus,iostat=ioStat,&
-      action=fAction,position=fPosition)
+      action=fAction,position=fPosition,err=10)
   endif
 
   if(ioStat /= 0) then
     err = .true.
     write(lout,"(a,i0)") "UNITS> File '"//trim(fFileName)//"' reported iostat = ",ioStat
-  else
-    err = .false.
   end if
   return
 
+10 continue
+  err = .true.
+  write(lout,"(a)") "UNITS> File '"//trim(fFileName)//"' reported an error"
```

Comment:
Why not `prror` if the err flag is not given? Also why not write to stderr instead/in addition to lout?

Reply:
Yes, but not for this PR.